### PR TITLE
Make sure that vested claims are not transferable

### PR DESCRIPTION
**Pull request by bkchr**
We need to deposit the claimed balance to the destination account before
setting the locks through the vesting module. Otherwise we loose the
locks and the vested claim is directly transferable.

### CODE REVIEW

File: runtime/common/src/claims.rs [hunk 1]
```diff
@@ -151,6 +151,8 @@ decl_error! {
 		PotUnderflow,
 		/// A needed statement was not included.
 		InvalidStatement,
+		/// The account already has a vested balance.
+		VestedBalanceExists,
```
Comment (shawntabrizi) [inline runtime/common/src/claims.rs:155]: You created a new event and now `DestinationVesting` is unused in the code

File: runtime/common/src/claims.rs [hunk 2]
```diff
@@ -438,15 +440,22 @@ impl<T: Trait> Module<T> {
 
 		let new_total = Self::total().checked_sub(&balance_due).ok_or(Error::<T>::PotUnderflow)?;
 
+		let vesting = Vesting::<T>::get(&signer);
+		if vesting.is_some() && T::VestingSchedule::vesting_balance(&dest).is_some() {
```
Comment (shawntabrizi) [inline runtime/common/src/claims.rs:444]: Should be or statement?

File: runtime/common/src/claims.rs (hunk 1 above)
Comment (bkchr) [reply runtime/common/src/claims.rs:155]: Is this one better? Or should I revert to `DestinationVesting`?

Comment (shawntabrizi) [reply runtime/common/src/claims.rs:155]: i mean, doesn't really matter to me. But should get rid of one and use the other :)

**Pull request merged**